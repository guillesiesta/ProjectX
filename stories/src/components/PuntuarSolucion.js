import React, {Component} from 'react';
import SolucionConCheck from '../components/SolucionConCheck'

export default class PuntuarSolucion extends Component {

  constructor(props){
    super(props);
    this.state={
      soluciones:[],
      storie:''
    };
  }

  componentDidMount(){
    //console.log("COSAS AQUI,titulo a buscar: "+this.props.titulo)
    fetch('https://projectx-eagwiugnbd.now.sh/soluciones_por_titulo', { //cargamos las soluciones del acertijo
          method: 'POST', // or 'PUT'
          body: JSON.stringify(this.props.titulo), // data can be `string` or {object}!
          headers:{
            'Content-Type': 'application/json'
          }})
      .then(response => response.json())
      .then(data => {
        //console.log(data)
        var array = [];
        var i;
        for(i=0; i<data.length;i++){
            let aux = {'puntuacion':data[i].puntuacion, 'solucion':data[i].solucion};
            array.push(aux);
            //console.log(data[i].puntuacion);
            //console.log(data[i].solucion);
          }
        this.setState({soluciones:array});
        //console.log("array length: "+array.length);
        /*var j;
        for(j=0; j<array.length;j++){
          console.log("Solución: "+array[j].solucion+" Puntuación: "+array[j].puntuacion);
        }*/
      })
      .catch(error => console.error(error))

      fetch('http://localhost:5000/storie_por_titulo', { //cargamos las soluciones del acertijo
            method: 'POST', // or 'PUT'
            body: JSON.stringify(this.props.titulo), // data can be `string` or {object}!
            headers:{
              'Content-Type': 'application/json'
            }})
        .then(response => response.json())
        .then(data => {
          //console.log("La storie es: "+data[0].storie)
          this.setState({storie:data[0].storie});
          //this.setState({storie:});
          //console.log("array length: "+array.length);
          /*var j;
          for(j=0; j<array.length;j++){
            console.log("Solución: "+array[j].solucion+" Puntuación: "+array[j].puntuacion);
          }*/
        })
        .catch(error => console.error(error))

  }

  verComentarios(){
    return(
      <div>
        {this.state.soluciones.map((i,index) => { //pongo index porque me obliga q ue las key sean diferentes
          //console.log("Solucion: "+i.solucion+" .Puntuación: "+i.puntuacion);
          return !this.state.sol_hidden && <SolucionConCheck key={index} solucion={i.solucion} puntuacion={i.puntuacion}/>
        })}
      </div>
    );
  //  console.log(this.state.soluciones[0].solucion)
  }
render(){

      return(
          <div className="box-header with-border">
            <div className="row">
              <strong>{this.props.titulo}</strong>
              <p>Solución propuesta por mi: <strong>{this.state.storie}</strong></p>
              <p>Soluciones propuestas por otros usuarios:</p>
              {this.verComentarios()}
            </div>
          </div>
      )
  }
}
