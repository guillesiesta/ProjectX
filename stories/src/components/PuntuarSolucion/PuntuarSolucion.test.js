import React, {Component} from 'react';
import { shallow } from 'enzyme';

import PuntuarSolucion from './PuntuarSolucion';

import "isomorphic-fetch"

describe('PuntuarSolucion', ()=> {
  test('Rendering the simple component', ()=>{

    const wrapper = shallow(
      <PuntuarSolucion />
    );

    expect(wrapper).toMatchSnapshot();
  })
})
