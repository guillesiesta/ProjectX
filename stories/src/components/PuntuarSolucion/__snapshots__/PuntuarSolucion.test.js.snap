// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PuntuarSolucion Rendering the simple component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PuntuarSolucion />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
      >
        <strong />
        <p>
          Solución propuesta por mi: 
          <strong>
            
          </strong>
        </p>
        <p>
          Soluciones propuestas por otros usuarios:
        </p>
        <div />
      </div>,
      "className": "box-header with-border",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <strong />,
          <p>
            Solución propuesta por mi: 
            <strong>
              
            </strong>
          </p>,
          <p>
            Soluciones propuestas por otros usuarios:
          </p>,
          <div />,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "strong",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Solución propuesta por mi: ",
              <strong>
                
              </strong>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Solución propuesta por mi: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "strong",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Soluciones propuestas por otros usuarios:",
          },
          "ref": null,
          "rendered": "Soluciones propuestas por otros usuarios:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <strong />
          <p>
            Solución propuesta por mi: 
            <strong>
              
            </strong>
          </p>
          <p>
            Soluciones propuestas por otros usuarios:
          </p>
          <div />
        </div>,
        "className": "box-header with-border",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong />,
            <p>
              Solución propuesta por mi: 
              <strong>
                
              </strong>
            </p>,
            <p>
              Soluciones propuestas por otros usuarios:
            </p>,
            <div />,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Solución propuesta por mi: ",
                <strong>
                  
                </strong>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Solución propuesta por mi: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": "strong",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Soluciones propuestas por otros usuarios:",
            },
            "ref": null,
            "rendered": "Soluciones propuestas por otros usuarios:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
